terraform/modules/sharded_module/preamble.tf

```diff
@@ -0,0 +1,52 @@
+resource "null_resource" "preamble" {
+  provisioner "local-exec" {
+    command = <<-EOT
+    sudo snap install juju-wait --classic || true
+    EOT
+  }
+}
+
+resource "juju_application" "self-signed-certificates" {
+  charm {
+    name    = "self-signed-certificates"
+    channel = "latest/stable"
+  }
+  model      = var.model_name
+  depends_on = [null_resource.preamble]
+}
+
+resource "juju_application" "data-integrator" {
+  charm {
+    name    = "data-integrator"
+    channel = "latest/stable"
+  }
+  model      = var.model_name
+  depends_on = [null_resource.preamble]
+}
```

Comment:
This will be related to mongos right?
How do we deploy without using data-integrator?

Reply:
> This will be related to mongos right?

Yes

> How do we deploy without using data-integrator?

We can't since mongos-k8s is a subordinate charm. We can use a client app, but thats out of scope here